src/components/Block.vue

```diff
@@ -425,27 +425,32 @@ function parseMarkdown (event:KeyboardEvent) {
   const textContent = getTextContent()
   if(!textContent) return
 
-  const headingRegexpMap = {
+  const markdownRegexpMap = {
     [BlockType.H1]: /^#\s(.*)$/,
     [BlockType.H2]: /^##\s(.*)$/,
     [BlockType.H3]: /^###\s(.*)$/,
+    [BlockType.Quote]: /^>\s(.*)$/,
+    [BlockType.Divider]: /^---$/
   }
-  const handleHeadingContent = (blockType: keyof typeof headingRegexpMap) => {
+
+  const handleMarkdownContent = (blockType: keyof typeof markdownRegexpMap) => {
     emit('setBlockType', blockType)
-    const newContent = textContent.replace(headingRegexpMap[blockType], '$1')
+    const newContent = textContent.replace(markdownRegexpMap[blockType], '$1')
     ;(content.value as any).innerText = newContent
     props.block.details.value = newContent
   }
 
-  if (textContent.match(headingRegexpMap[BlockType.H1]) && event.key === ' ') {
-    handleHeadingContent(BlockType.H1)
-  } else if (textContent.match(headingRegexpMap[BlockType.H2]) && event.key === ' ') {
-    handleHeadingContent(BlockType.H2)
-  } else if (textContent.match(headingRegexpMap[BlockType.H3]) && event.key === ' ') {
-    handleHeadingContent(BlockType.H3)
-  } else if (textContent.match(/^---$/)) {
-    emit('setBlockType', BlockType.Divider);
-    (content.value as any).innerText = ''
+
+  if (textContent.match(markdownRegexpMap[BlockType.H1]) && event.key === ' ') {
+    handleMarkdownContent(BlockType.H1)
+  } else if (textContent.match(markdownRegexpMap[BlockType.H2]) && event.key === ' ') {
+    handleMarkdownContent(BlockType.H2)
+  } else if (textContent.match(markdownRegexpMap[BlockType.H3]) && event.key === ' ') {
+    handleMarkdownContent(BlockType.H3)
+  } else if (textContent.match(markdownRegexpMap[BlockType.Quote]) && event.key === ' ') {
+    handleMarkdownContent(BlockType.Quote)
+  } else if (textContent.match(markdownRegexpMap[BlockType.Divider])) {
+    handleMarkdownContent(BlockType.Divider)
```

Comment:
Thanks for the quick update!

One small fix - can you also add
`props.block.details.value = ''`
after this (at Line 454 inside the matched else-if for Divider), otherwise the value for the block will be set to `"$1"` when converting to a Divider.

Reply:
Sorry, my bad

Reply:
LGTM! 🧴